NuclearData/mgIMCData/baseMgIMC/baseMgIMCMaterial_class.f90

```diff
@@ -0,0 +1,487 @@
+    call kill_super(self)
+
+    ! Kill local content
+    if(allocated(self % data)) deallocate(self % data)
+
+  end subroutine kill
+
+  !!
+  !! Load Macroscopic XSs into the provided package for a given group index G
+  !!
+  !! See mgIMCMaterial documentation for more details
+  !!
+  subroutine getMacroXSs_byG(self, xss, G, rand)
+    class(baseMgIMCMaterial), intent(in)     :: self
+    type(IMCMacroXSs), intent(out)           :: xss
+    integer(shortInt), intent(in)            :: G
+    class(RNG), intent(inout)                :: rand
+    character(100), parameter :: Here = ' getMacroXSs (baseMgIMCMaterial_class.f90)'
+
+    ! Verify bounds
+    if(G < 1 .or. self % nGroups() < G) then
+      call fatalError(Here,'Invalid group number: '//numToChar(G)// &
+                           ' Data has only: ' // numToChar(self % nGroups()))
+    end if
+
+    ! Get XSs
+    xss % total            = self % data(TOTAL_XS, G)
+    xss % elasticScatter   = ZERO
+    xss % inelasticScatter = self % data(IESCATTER_XS, G)
+    xss % capture          = self % data(CAPTURE_XS, G)
+    xss % planck           = self % data(PLANCK_XS, G)
+
+  end subroutine getMacroXSs_byG
+
+  !!
+  !! Return Total XSs for energy group G
+  !!
+  !! See mgIMCMaterial documentationfor details
+  !!
+  function getTotalXS(self, G, rand) result(xs)
+    class(baseMgIMCMaterial), intent(in)     :: self
+    integer(shortInt), intent(in)            :: G
+    class(RNG), intent(inout)                :: rand
+    real(defReal)                            :: xs
+    character(100), parameter :: Here = ' getTotalXS (baseMgIMCMaterial_class.f90)'
+
+    ! Verify bounds
+    if(G < 1 .or. self % nGroups() < G) then
+      call fatalError(Here,'Invalid group number: '//numToChar(G)// &
+                           ' Data has only: ' // numToChar(self % nGroups()))
+      xs = ZERO ! Avoid warning
+    end if
+    xs = self % data(TOTAL_XS, G)
+
+  end function getTotalXS
+
+
+  !!
+  !! Initialise Base MG IMC Material fromdictionary
+  !!
+  !! Args:
+  !!   dict       [in] -> Input dictionary with all required XSs
+  !!
+  !! Errors:
+  !!   FatalError if data in dictionary is invalid (inconsistant # of groups;
+  !!     -ve entries in P0 XSs)
+  !!
+  subroutine init(self, dict)
+    class(baseMgIMCMaterial), intent(inout)     :: self
+    class(dictionary),target, intent(in)        :: dict
+    integer(shortInt)                           :: nG, N
+    real(defReal), dimension(:), allocatable    :: temp
+    character(100), parameter :: Here = 'init (baseMgIMCMaterial_class.f90)'
+
+    ! Read number of groups
+    call dict % get(nG, 'numberOfGroups')
+    if(nG < 1) call fatalError(Here,'Number of groups is invalid' // numToChar(nG))
+
+    ! Allocate space for data
+    N = 4
+    allocate(self % data(N, nG))
+
+    ! Store alpha setting
+    call dict % getOrDefault(self % alpha, 'alpha', ONE)
+
+    ! Read opacity equations
+    call dict % get(temp, 'sigmaA')
```

Comment:
Actually, I missed this first time round. It would better correspond to the other MG implementations if these were called 'capture' and 'scatter' instead in the input data file. Sorry I didn't spot this sooner!